packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -1,67 +1,378 @@
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if data is undefined', () => {
-    const event = { data: undefined } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if data is null', () => {
-    const event = { data: null } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('shouldProcessMessage fail if message source is same window ', () => {
-    communication.Communication.currentWindow = window;
-    // window object should now equal Communication.currentWindow
-    const result = communication.shouldProcessMessage(window, 'testOrigin.com');
-    expect(result).toBe(false);
+      beforeEach(() => {
+        // Set a mock window for testing
+        utils = new Utils();
+        utils.mockWindow.parent = undefined;
+        app._initialize(utils.mockWindow);
+        communication.Communication.parentWindow = undefined;
+        GlobalVars.isFramelessWindow = false;
+      });
+
+      afterAll(() => {
+        communication.uninitializeCommunication();
+        GlobalVars.isFramelessWindow = false;
+      });
+
+      it('should throw if there is no parent window and no native interface on the current window', async () => {
+        app._initialize(undefined);
+        const initPromise = communication.initializeCommunication(undefined);
+        await expect(initPromise).rejects.toThrowError('Initialization Failed. No Parent window found.');
+      });
+
+      it('should receive valid initialize response from parent', async () => {
+        const initPromise = communication.initializeCommunication(undefined);
+        const initMessage = utils.findInitializeMessageOrThrow();
+
+        utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+        const initializeResponse = await initPromise;
+
+        const expectedResponse = {
+          context: FrameContexts.content,
+          clientType: undefined,
+          runtimeConfig: undefined,
+          clientSupportedSDKVersion: undefined,
+        };
+        expect(initializeResponse).toStrictEqual(expectedResponse);
+      });
+
+      it('Communication.currentWindow should be unchanged by initializeCommunication', async () => {
+        expect(communication.Communication.currentWindow).toStrictEqual(utils.mockWindow);
+
+        const initPromise = communication.initializeCommunication(undefined);
+        const initMessage = utils.findInitializeMessageOrThrow();
+
+        utils.respondToNativeMessage(initMessage, false, FrameContexts.content);
+
+        await initPromise;
+
+        expect(communication.Communication.currentWindow).toStrictEqual(utils.mockWindow);
+      });
+
+      it('should set Communication.parentWindow to undefined', async () => {
```

Comment:
>should set Communication.parentWindow to undefined [](http://example.com/codeflow?start=10&length=50)

I think it might be helpful if some of these unit test descriptions had just a little more info to aid readers. E.g. after thinking about this test for a bit I think what would have helped me if the test description was something like

"should set Communication.parentWindow to be undefined because provided window parent is undefined"

and the one below this

"should set window.onNativeMessage for handling responses because..."

(Maybe you could argue that these "because" clauses should be in the code themselves, and I agree that's true, but it doesn't feel harmful to provide a bit more detail in the unit test description? I know we haven't done a lot of this in this code base, but typically I think the descriptive unit tests include both the expect result AND the state/scenario causing that result, e.g. MethodName_ExpectedBehavior_StateUnderState, like InitializeCommunication_SetsParentWindowToUndefined_WhenGivenWindowParentIsUndefined)

Reply:
Oh I see now that you've start to do a bit of that in the test on line 90!

Reply:
I will see what I can do, I do agree that given the complexity of this area I'd like these unit tests to be as descriptive as possible about what they're testing and why